authorization/authorization.go

```diff
@@ -0,0 +1,161 @@
+package authorization
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"log"
+	"net/http"
+	"time"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+type Handshake struct {
+	nonce      string
+	providerID string
```

Comment:
"providerID" makes an assumption about the type of entity being authorized, but this module aims to be generic. 

Moreover, I'm not sure abstracting this into a separate module is right just yet. I might wait until we've nailed down the API on the provider side. Then we can see whether this can be made generic.

Reply:
Had the exact same thought in regards to the naming convention, however I don't see any issues with a seperate module.

Reply:
Missed that. But, a handshake, for the purposes of the module, should never need to be anything more than an ID mapped to a nonce (and possibly some expiration time or other data).